src/main/java/frc/robot/subsystems/flywheel/Flywheel.java

```diff
@@ -0,0 +1,269 @@
+// Copyright FRC 5712
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// version 3 as published by the Free Software Foundation or
+// available in the root directory of this project.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+package frc.robot.subsystems.flywheel;
+
+import static edu.wpi.first.units.Units.*;
+
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Voltage;
+import edu.wpi.first.wpilibj.Alert;
+import edu.wpi.first.wpilibj.Alert.AlertType;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SelectCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import java.util.Map;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+/**
+ * The Flywheel subsystem controls a dual-motor flywheel mechanism used for shooting game pieces. It
+ * supports multiple shooting modes with different target speeds and provides both open-loop and
+ * closed-loop control options.
+ */
+public class Flywheel extends SubsystemBase {
+  // Hardware interface and inputs
+  private final FlywheelIO io;
+  private final FlywheelIOInputsAutoLogged inputs;
+
+  // Alerts for motor connection status
+  private final Alert leaderMotorAlert =
```

Comment:
what is intersting... here.. is that these chunks of code are identical to the arm subsystem. 
maybe there is opportunity to refactor this to have the subsystem classes either extend another class or , better, compose them so that the sysId and the Alerting can all be "registered" .. so this code coudl say like:
AlertRegister.registerAlert("flywheel isn't connected",() -> {}); (a palce for a potential call back method) maybe you do it like the optionals.. ie registerAlert(String message, Function success, Function failure)